endercore/src/main/java/com/enderio/core/client/tooltip/TooltipHandler.java

```diff
@@ -0,0 +1,63 @@
+package com.enderio.core.client.tooltip;
+
+import com.enderio.core.common.lang.EnderCoreLang;
+import net.minecraft.ChatFormatting;
+import net.minecraft.client.gui.screens.Screen;
+import net.minecraft.network.chat.Component;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.BlockItem;
+import net.minecraft.world.item.ItemStack;
+import net.minecraftforge.api.distmarker.Dist;
+import net.minecraftforge.event.entity.player.ItemTooltipEvent;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import net.minecraftforge.fml.common.Mod;
+
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.List;
+
+@Mod.EventBusSubscriber(value = Dist.CLIENT, bus = Mod.EventBusSubscriber.Bus.FORGE)
+public class TooltipHandler {
+
+    private static final List<Component> tempTooltip = new ArrayList<>();
+
+    @SubscribeEvent
+    public static void addTooltips(ItemTooltipEvent evt) {
+        ItemStack forItem = evt.getItemStack();
+        IAdvancedTooltipProvider provider = null;
+        if(forItem.getItem() instanceof IAdvancedTooltipProvider p) {
+            provider = p;
+        } else if(forItem.getItem() instanceof BlockItem bi && bi.getBlock() instanceof  IAdvancedTooltipProvider p) {
+            provider = p;
+        }
+        if(provider != null) {
+            addTooltips(provider, forItem, evt.getPlayer(), evt.getToolTip(), shouldShowAdvancedTooltips());
+        }
+    }
+
+    public static void addTooltips(IAdvancedTooltipProvider tt, ItemStack itemstack, @Nullable Player player, List<Component> list, boolean showAdvanced) {
+        tt.addCommonTooltips(itemstack, player, list);
+        if (showAdvanced) {
+            tt.addDetailedTooltips(itemstack, player, list);
+        } else {
+            tt.addBasicTooltips(itemstack, player, list);
+            if (hasDetailedTooltip(tt, itemstack, player)) {
```

Comment:
Would it be possible to cut this down, we're first querying whether there is a detailed tooltip (which creates it) and then we ask for it to be created again when adding it. Is there a way of doing this in one call to the detailed tooltip? Just a minor efficiency comment but certainly not a problem if it gets too complicated.

Reply:
We don't use the detailed tooltips here, we add the <Hold Shift> tooltip and throw away the detailed ones

Reply:
Sounds good to me